make/hotspot/lib/JvmOverrideFiles.gmk

```diff
@@ -147,6 +148,55 @@ else ifeq ($(call isTargetOs, windows), true)
   # Workaround for jvmciCompilerToVM.cpp long compilation time
   BUILD_LIBJVM_jvmciCompilerToVMInit.cpp_OPTIMIZATION := NONE
 
+else ifeq ($(call isTargetOs, bsd), true)
+  JVM_EXCLUDE_FILES += decoder_mach0.cpp
+  BUILD_LIBJVM_sharedRuntimeTrig.cpp_CXXFLAGS := -DNO_PCH $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
+  BUILD_LIBJVM_sharedRuntimeTrans.cpp_CXXFLAGS := -DNO_PCH $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
+
+  BUILD_LIBJVM_loopTransform.cpp_CXXFLAGS := $(CXX_O_FLAG_NONE)
+  ifneq ($(DEBUG_LEVEL), slowdebug)
+    BUILD_LIBJVM_unsafe.cpp_CXXFLAGS := -O1
+  endif
+
+  ifeq ($(TOOLCHAIN_TYPE), gcc)
+    BUILD_LIBJVM_unsafe.cpp_CXXFLAGS += -Wno-maybe-uninitialized
+  endif
+
+  ifeq ($(TOOLCHAIN_TYPE), clang)
+    ifeq ($(call isTargetCpu, x86), true)
+      ifneq ($(DEBUG_LEVEL), slowdebug)
+        # hotspot/jtreg/compiler/c2/Test8062950.java test fails on x86
+        # with clang when parse2.cpp is optimized above -O1
+        BUILD_LIBJVM_parse2.cpp_CXXFLAGS := -O1
+      endif
+    endif
+
+    # The following files are compiled at various optimization
+    # levels due to optimization issues encountered at the
+    # default level. The Clang compiler issues a compile
+    # time error if there is an optimization level specification
+    # skew between the PCH file and the C++ file.  Especially if the
+    # PCH file is compiled at a higher optimization level than
+    # the C++ file.  One solution might be to prepare extra optimization
+    # level specific PCH files for the opt build and use them here, but
+    # it's probably not worth the effort as long as only a few files
+    # need this special handling.
+    JVM_PRECOMPILED_HEADER_EXCLUDE := \
```

Comment:
There have been some recent changes in the corresponding piece of code for other OSes, you might want to check how this is done in the latest mainline.

Reply:
Yes, I have seen that. Already made a note on going through to see if any of these changes actually are necessary. Now I have an extra note :)